proxy/proxy_test.go

```diff
@@ -327,38 +376,76 @@ func startBackendGRPC(grpcServer *grpc.Server) error {
 
 // genCertPair generates a pair of private key and certificate and returns them
 // in different formats needed to spin up test servers and clients.
-func genCertPair(certFile, keyFile string) (*x509.CertPool,
-	credentials.TransportCredentials, tls.Certificate, error) {
+func genCertPair(certFile, keyFile string) (*tls.Config,
+	credentials.TransportCredentials, error) {
 
-	crt := tls.Certificate{}
 	err := cert.GenCertPair(
 		"aperture autogenerated cert", certFile, keyFile, nil, nil,
-		cert.DefaultAutogenValidity,
+		false, cert.DefaultAutogenValidity,
 	)
 	if err != nil {
-		return nil, nil, crt, fmt.Errorf("unable to generate cert "+
+		return nil, nil, fmt.Errorf("unable to generate cert "+
 			"pair: %v", err)
 	}
 
 	crt, x509Cert, err := cert.LoadCert(certFile, keyFile)
 	if err != nil {
-		return nil, nil, crt, fmt.Errorf("unable to load cert: %v", err)
+		return nil, nil, fmt.Errorf("unable to load cert: %v", err)
 	}
 
 	cp := x509.NewCertPool()
 	cp.AddCert(x509Cert)
 
 	creds, err := credentials.NewClientTLSFromFile(certFile, "")
 	if err != nil {
-		return nil, nil, crt, fmt.Errorf("unable to load cert file: "+
+		return nil, nil, fmt.Errorf("unable to load cert file: "+
 			"%v", err)
 	}
-	return cp, creds, crt, nil
+
+	tlsConf := cert.TLSConfFromCert(crt)
+	tlsConf.CipherSuites = []uint16{
+		tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+	}
+	tlsConf.PreferServerCipherSuites = true
+	tlsConf.RootCAs = cp
+	tlsConf.InsecureSkipVerify = true
+
+	haveNPN := false
+	for _, p := range tlsConf.NextProtos {
+		if p == "h2" {
+			haveNPN = true
+			break
+		}
+	}
+	if !haveNPN {
+		tlsConf.NextProtos = append(tlsConf.NextProtos, "h2")
+	}
+	tlsConf.NextProtos = append(tlsConf.NextProtos, "h2-14")
```

Comment:
Would this serve to resolve those issues we observed when we attempted to expose Aperture _directly_ over the load balancer?